### ISSUE TITLE
How to receive more XMPP parameters

### ISSUE THREAD

**Issue by alfredogt** [jitsi/jibri#40]
Hello everyone,
I had some disadvantages in making jibri work in a more dynamic way.

I want to help improve JIBRI

Currently jibri, does not receive the URL for the recording, nor does it receive the URL of the RTMP server.
This makes jibri quite limited, especially to make independent implementations.

My concern is: Is it possible to send more parameters when sending the message to Jibri?

I did tests sending different parameters, but jibri only those specified in this documentation:
Https://github.com/jitsi/jibri/blob/master/jibri-spec.xml

If this were possible, we could improve a lot of jibri to send transmissions to FacebookLive, to different RTMP servers, we could start FFMPG with multiple outputs:
Https://trac.ffmpeg.org/wiki/Creating%20multiple%20outputs

In short, many possibilities.

In order to make jibri more flexible.

**Comment by aaronkvanmeerten** [comment]
I'm interested in your use cases here for multiple outputs.  Would you expect to stream to both youtube and facebook in the same stream?
The issue with adding more parameters to the XMPP messages is mostly about instrumenting them throughout jitsi-meet, lib-jitsi-meet and jicofo.  Each of these components are involved one way or another when starting a jibri.  We are always open to contributions on each of these components, and are open to discussing the types of changes you wish to make and making recommendations based on our current experience.  We are slowly improving this protocol as we go, but haven't considered the multiple outputs scenario at all yet.  Can you tell us more about this use case?

**Comment by alfredogt** [comment]
> Would you expect to stream to both youtube and facebook in the same stream?
YES, that would be a possible use case

I could give some examples of interesting use cases.
My idea is that in the future JITSI be oriented more to business and commercial use.

I will try to explain each of the use cases, from my perspective.

Use cases:
* Send the transmission to facebook LIVE
 The use of Facebook Live is currently taking a lot of force from the commercial point of view, I have not found in the market any platform that allows me to have a meeting of many people and that allows me to transmit to a RTMP server (Jitsi is the only FREE) , this is a great advantage.
I think FaceBook LIVE is necessary.

* Send the transmission to a specific RTMP.
 Currently, there are many services for retransmitting RTMP, for example: Kaltura, nginx-rtmp, nimbleStream ... and many more.
From the business point of view, I think it is important to give the possibility of using any RTMP destination.
I know this is going to represent some challenges for the different types of
 configuration that each service can have. But I think it could be worth it.
I think having many RTMP destinations is going to make JITSI more interesting for the public.

Felibitability in JIBRI.
I have created my own implementation, using lib-jitsi-meet, but Jibri is very limited.
For example:

My URL must be strictly: https://midominio.com/namemyroom
In this way jibri is built to connect to the room, but it is very limiting for different implementations, for example if I want to have a specific URL for each room:
For example:
https://mydomain.com/paramx/valory/nameroom/iduser
or
https://mydomain.com?nameroom=mynameroom


Currently the use of streaming has a lot of strength, many systems like facebook, youtube and others are putting many efforts to do it better and better.
This is good!
I think Jitsi-Meet, should focus on this issue, the LIVE broadcast.

_______________________________________________________________________________________________
How do I propose solution: send a single additional parameter in each XMMP message (Optional).
We send a JSON with all the necessary configuration to be able to configure jibri depending on my needs.
In this way I think it is more flexible and we would not have to make many changes to jitsi-meet and lib-jitsi-meet and jicofo.

If we receive a JSON in jibri, we would have the possibility to create changes in a very flexible way.

I'm not very familiar with XMPP, so I'm not sure if this form affects the protocol or something I'm saying something absurd hehe

I hope comments ...

Greetings.

(My English is very bad, I'm sorry)

**Comment by aaronkvanmeerten** [comment]
I would very much recommend using the latest java-based jibri from our debian respository.  It's easier to use and is much more reliable about getting connected.  I'm going to close this issue, but please open a new one if you run into any problems with the new java-based jibri!
